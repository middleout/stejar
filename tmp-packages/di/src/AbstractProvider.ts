import { ServiceManager } from "./ServiceManager";

export abstract class AbstractProvider<T> {
    /**
     * @type {typeof T}
     */
    abstract provides(): { new (...args: any[]): T } | string;

    /**
     * @param serviceManager
     * @returns {T}
     */
    abstract provide(serviceManager: ServiceManager): T;
}
