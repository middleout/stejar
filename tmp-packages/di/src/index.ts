export * from "./ServiceManager";
export * from "./AbstractProvider";
export * from "./injectable";