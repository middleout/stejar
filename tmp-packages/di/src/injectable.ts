/**
 * @param WrappedClass
 * @returns {T}
 */
export function injectable<T>(WrappedClass: T): T {
    return WrappedClass;
}
