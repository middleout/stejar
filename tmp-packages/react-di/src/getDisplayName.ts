/**
 * @param WrappedComponent
 * @returns {string|string}
 */
export function getDisplayName(WrappedComponent: any): string {
    return WrappedComponent.displayName || WrappedComponent.name || "AnonymousComponent";
}