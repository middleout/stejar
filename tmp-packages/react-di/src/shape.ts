import PropTypes from "prop-types";

export const shape = PropTypes.object;
