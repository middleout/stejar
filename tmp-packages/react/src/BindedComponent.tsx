import { Component as ReactComponent } from "react";
import { binder } from "./binder";

export abstract class BindedComponent<P = {}, S = {}> extends ReactComponent<P, S> {
    props: P & { children?: any };

    /**
     * @param props
     * @param context
     */
    constructor(props: P, context?: any) {
        super(props, context);
        binder(this);
    }
}
