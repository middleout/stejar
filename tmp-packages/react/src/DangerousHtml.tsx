import { HTMLProps } from "react";
import DOM from "react-dom-factories";

export interface DangerousHtmlProps extends HTMLProps<any> {
    element?: string | "span";
}

/**
 * @param props
 * @returns {any}
 * @constructor
 */
export function DangerousHtml(props: DangerousHtmlProps) {
    const newProps: any = { ...props };
    (newProps as any).dangerouslySetInnerHTML = {
        __html: props.children as string,
    };
    delete newProps.children;
    const element = newProps.element;
    delete newProps.element;
    if (!(newProps as any).className) {
        (newProps as any).className = "dangerousHtml";
    } else {
        (newProps as any).className += " dangerousHtml";
    }
    return DOM[element || "span"](newProps);
}
