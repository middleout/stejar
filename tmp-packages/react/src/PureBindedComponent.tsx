import { PureComponent as ReactPureComponent } from "react";
import { binder } from "./binder";

export abstract class PureBindedComponent<P = {}, S = {}> extends ReactPureComponent<P, S> {
    props: P & { children?: any };

    /**
     * @param props
     * @param context
     */
    constructor(props: P, context?: any) {
        super(props, context);
        binder(this);
    }
}
