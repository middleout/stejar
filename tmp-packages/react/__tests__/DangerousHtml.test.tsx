import { render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { DangerousHtml } from "../src/DangerousHtml";

test("Should be able to use it", () => {
    const component = renderer.create(<DangerousHtml>Hello World</DangerousHtml>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const dom = render(<DangerousHtml>Hello World</DangerousHtml>);
    expect(dom.find('span').text()).toEqual("Hello World");
    expect(dom.find('span').attr('class')).toEqual("dangerousHtml");
});

test("Should be able to use it by giving it a class", () => {
    const component = renderer.create(<DangerousHtml className="foo">Hello World</DangerousHtml>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const dom = render(<DangerousHtml className="foo">Hello World</DangerousHtml>);
    expect(dom.find('span').text()).toEqual("Hello World");
    expect(dom.find('span').attr('class')).toEqual("foo dangerousHtml");
});

test("Should be able to use it by giving it a differnet element", () => {
    const component = renderer.create(<DangerousHtml element="div">Hello World</DangerousHtml>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const dom = render(<DangerousHtml element="div">Hello World</DangerousHtml>);
    expect(dom.find('div').text()).toEqual("Hello World");
});
