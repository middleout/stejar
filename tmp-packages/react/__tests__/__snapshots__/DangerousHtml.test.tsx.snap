// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to use it 1`] = `
<span
  className="dangerousHtml"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Hello World",
    }
  }
/>
`;

exports[`Should be able to use it by giving it a class 1`] = `
<span
  className="foo dangerousHtml"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Hello World",
    }
  }
/>
`;

exports[`Should be able to use it by giving it a differnet element 1`] = `
<div
  className="dangerousHtml"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Hello World",
    }
  }
/>
`;
